src/Service/Player/Url.php

```diff
@@ -123,10 +138,19 @@ public function __toString()
      * @see https://docs.theplatform.com/help/player-player-autoplay
      *
      * @param bool $autoPlay True to enable autoPlay, false otherwise.
+     *
+     * @return Url
      */
-    public function setAutoplay(bool $autoPlay)
+    public function withAutoplay(bool $autoPlay): self
     {
-        $this->autoPlay = $autoPlay;
+        if ($this->autoPlay == $autoPlay) {
+            return $this;
+        }
+
+        $url = clone $this;
+        $url->autoPlay = $autoPlay;
```

Comment:
Apparently you can modify private properties when an object clones itself. 🍌 !

http://php.net/manual/en/language.oop5.cloning.php#122822

Reply:
Huh, look at that.